libsolidity/ast/ASTJsonImporter.cpp

```diff
@@ -348,7 +353,17 @@ ASTPointer<ContractDefinition> ASTJsonImporter::createContractDefinition(Json co
 		baseContracts,
 		subNodes,
 		contractKind(_node),
-		memberAsBool(_node, "abstract")
+		memberAsBool(_node, "abstract"),
+		nullOrCast<StorageLayoutSpecifier>(member(_node, "storageLayoutSpecifier"))
+	);
+}
+
+ASTPointer<StorageLayoutSpecifier> ASTJsonImporter::createStorageLayoutSpecifier(Json const& _node)
+{
+	astAssert(_node.contains("expression"));
```

Comment:
Note that asserts in the importer are generally abused for validation. For this reason they should all have messages because otherwise the user will only see a generic assertion message.

But it would really be better to define an exception type for the importer and start using `solRequire()` here.

Reply:
I think we should add the exception type, but on a different PR. Then we can refactor all the validation asserts.